analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MagicNumberShouldNotBeUsed.cs

```diff
@@ -27,10 +28,20 @@ public ValidUseCases()
             }
 
             const int VAL = 15;
-
+            
             Console.Write("test");
         }
 
+        /// <summary>
+        ///     A five minute sliding expiration 
+        /// </summary>
+        public static CacheItemPolicy FiveMinSlide => new CacheItemPolicy {SlidingExpiration = new TimeSpan(0, 5, 0)};
```

Comment:
indeed DateTime APIs could be considered as well-known cases. The problem is how to avoid having to maintain such a list, we would need to find something generic enough.

Reply:
if parameter name is the name of a specific time measurement ("day" "month" "year", etc), then the usage of a literal has an obvious, specific, and easy to understand meaning and is thus not "magic".

Reply:
The same would be true for "Order"

Reply:
May be first go back to the drawing board, and think of what to support first?

Reply:
@Corniel  as a user of sonarcloud, I want it to report when a literal number has been used in a place where its not obvious what that number's meaning is.

Reply:
@StingyJack I got that point. This a pull request. I saw that @andrei-epure-sonarsource already made an issue to discuss the fllow up: #4737

Reply:
Then I'm not sure what you're looking for here from me

Reply:
@StingyJack Well, I'm just a volunteer, but I would think that your input on what should not be considered magical numbers would help. As @andrei-epure-sonarsource andrei-epure-sonarsource mentioned, the challenge is to come with generic constraints, instead of some types that should be excluded, as the maintenance cost is high for those.

Reply:
I initially asked for a PR with test cases on the community forum, hence this PR.
I will close this PR in  favor of #4737.
Thanks @StingyJack for the discussion and this PR.